tests/Aspire.Dashboard.Components.Tests/Layout/MainLayoutTests.cs

```diff
@@ -125,7 +125,61 @@ public async Task OnInitialize_UnsecuredOtlp_Dismissed_NoMessageBar()
         Assert.Empty(messageService.AllMessages);
     }
 
-    private void SetupMainLayoutServices(TestLocalStorage? localStorage = null, MessageService? messageService = null)
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task OnInitialize_UnsecuredOtlp_SuppressConfigured_NoMessageBar(bool suppressUnsecuredMessage)
+    {
+        // Arrange
+        var testLocalStorage = new TestLocalStorage();
+        var messageService = new MessageService();
+
+        SetupMainLayoutServices(localStorage: testLocalStorage, messageService: messageService, suppressUnsecuredMessage: suppressUnsecuredMessage);
+
+        var messageShownTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        messageService.OnMessageItemsUpdatedAsync += () =>
+        {
+            messageShownTcs.TrySetResult();
+            return Task.CompletedTask;
+        };
+
+        testLocalStorage.OnGetUnprotectedAsync = key =>
+        {
+            if (key == BrowserStorageKeys.UnsecuredTelemetryMessageDismissedKey)
+            {
+                return (false, false); // Message not dismissed, but should be suppressed by config if suppressUnsecuredMessage is true
+            }
+            else
+            {
+                throw new InvalidOperationException("Unexpected key.");
+            }
+        };
+
+        // Act
+        var cut = RenderComponent<MainLayout>(builder =>
+        {
+            builder.Add(p => p.ViewportInformation, new ViewportInformation(IsDesktop: true, IsUltraLowHeight: false, IsUltraLowWidth: false));
+        });
+
+        // Assert
+        var timeoutTask = Task.Delay(100);
+        var completedTask = await Task.WhenAny(messageShownTcs.Task, timeoutTask).WaitAsync(TimeSpan.FromSeconds(5));
+
+        if (suppressUnsecuredMessage)
+        {
+            // When suppressed, no message should be displayed
+            Assert.True(completedTask != messageShownTcs.Task, "No message bar should be displayed when suppressed by configuration.");
+            Assert.Empty(messageService.AllMessages);
+        }
+        else
+        {
+            // When not suppressed, message should be displayed since it wasn't dismissed
+            Assert.True(completedTask == messageShownTcs.Task, "Message bar should be displayed when not suppressed and not dismissed.");
+            Assert.NotEmpty(messageService.AllMessages);
+        }
```

Comment:
```suggestion
        if (suppressUnsecuredMessage)
        {
            var timeoutTask = Task.Delay(100);
            var completedTask = await Task.WhenAny(messageShownTcs.Task, timeoutTask).DefaultTimeout();

            // When suppressed, no message should be displayed
            Assert.True(completedTask != messageShownTcs.Task, "No message bar should be displayed when suppressed by configuration.");
            Assert.Empty(messageService.AllMessages);
        }
        else
        {
            // When not suppressed, message should be displayed since it wasn't dismissed
            await messageShownTcs.Task.DefaultTimeout();
            Assert.NotEmpty(messageService.AllMessages);
        }
```